components/availability/src/Availability.svelte

```diff
@@ -247,6 +253,19 @@
           availability = AvailabilityStatus.BUSY;
         }
 
+        if (
+          availability === AvailabilityStatus.PARTIAL &&
+          required_participants.length
+        ) {
+          if (
+            !required_participants.every((participant) =>
+              freeCalendars.includes(participant),
+            )
+          ) {
+            availability = AvailabilityStatus.BUSY;
+          }
+        }
+
```

Comment:
Is this nested `if` block needed? Why not 
```
if (availability === AvailabilityStatus.PARTIAL && required_participants.length && !required_participants.every((participant) => freeCalendars.includes(participant))) {
  availability = AvailabilityStatus.BUSY;
}
```

Reply:
This was originally written like this for performance sake: .every() is a potentially expensive calculation and I only want to do it on those timeSlots that are already partial, but now that you mention it, I think that I can `&&` chain and it won't evaluate later items if prior conditions are falsy.

Still -- my feeling is that doing this in a nested way emphasizes the importance of doing this check only if conditions are met. I'm not sure if the one-liner conditional is as clear.